graph/path/yen_ksp.go

```diff
@@ -0,0 +1,130 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package path
+
+import (
+	"sort"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/internal/set"
+)
+
+type yenShortest struct {
+	p   []graph.Node
+	weight float64
+}
+
+func YenKSP(source graph.Node, sink graph.Node, g graph.Weighted, k int) [][]graph.Node {
```

Comment:
Conventionally we use `s` and `t`, and I think YenKSP as a name could be improved on: `YenKShortestPaths(g graph.Graph, k int, s, t graph.Node) [][]graph.Node`.

The weighting should be determined dynamically using unit cost in the case that `g` is not a `graph.Weighted`.

Reply:
How does uniform cost work? Currently, I am adding up the root weight to each computed spur weight; is it possible to do that using uniform cost?

Reply:
Each edge just has a weight of 1. If you handed in a weighted graph where each edge had unit weight, that would be the same.

In practice, make a `Weighting` (`func(xid, yid int64) (w float64, ok bool)`) and assign it either the `g.Weight` method or `UniformCost(g)` like so:
```
	var weight Weighting
	if wg, ok := g.(Weighted); ok {
		weight = wg.Weight
	} else {
		weight = UniformCost(g)
	}
```

Then use `weight` to determine weights where you currently use `g.Weight`.